backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/service/OwnDeliveryService.java

```diff
@@ -0,0 +1,136 @@
+                .toList();
+    }
+
+    public final List<OwnDelivery> findAllByOwnMaterialNumber(String ownMaterialNumber) {
+        return repository.findAll().stream().filter(delivery -> delivery.getMaterial().getOwnMaterialNumber().equals(ownMaterialNumber))
+                .toList();
+    }
+
+    public final List<OwnDelivery> findAllByFilters(Optional<String> ownMaterialNumber, Optional<String> bpnl) {
+        Stream<OwnDelivery> stream = repository.findAll().stream();
+        if (ownMaterialNumber.isPresent()) {
+            stream = stream.filter(delivery -> delivery.getMaterial().getOwnMaterialNumber().equals(ownMaterialNumber.get()));
+        }
+        if (bpnl.isPresent()) {
+            stream = stream.filter(delivery -> delivery.getPartner().getBpnl().equals(bpnl.get()));
+        }
+        return stream.toList();
+    }
+
+    public final OwnDelivery findById(UUID id) {
+        return repository.findById(id).orElse(null);
+    }
+
+    public final OwnDelivery create(OwnDelivery delivery) {
+        if (!validator.apply(delivery)) {
+            throw new IllegalArgumentException("Invalid delivery");
+        }
+        if (delivery.getUuid() != null && repository.findById(delivery.getUuid()).isPresent()) {
+            throw new KeyAlreadyExistsException("Delivery already exists");
+        }
+        return repository.save(delivery);
+    }
+
+    public final List<OwnDelivery> createAll(List<OwnDelivery> deliveries) {
+        if (deliveries.stream().anyMatch(delivery -> !validator.apply(delivery))) {
+            throw new IllegalArgumentException("Invalid delivery");
+        }
+        if (repository.findAll().stream()
+                .anyMatch(existing -> deliveries.stream().anyMatch(delivery -> delivery.equals(existing)))) {
+            throw new KeyAlreadyExistsException("delivery already exists");
+        }
+        return repository.saveAll(deliveries);
+    }
+
+    public final OwnDelivery update(OwnDelivery delivery) {
+        if (delivery.getUuid() == null || repository.findById(delivery.getUuid()).isEmpty()) {
+            return null;
+        }
+        return repository.save(delivery);
+    }
+
+    public final void delete(UUID id) {
+        repository.deleteById(id);
+    }
+
+    public boolean validate(OwnDelivery delivery) {
+        Partner ownPartnerEntity = partnerService.getOwnPartnerEntity();
+        return 
+            delivery.getQuantity() > 0 && 
+            delivery.getMeasurementUnit() != null &&
+            delivery.getMaterial() != null &&
+            delivery.getPartner() != null &&
+            delivery.getTrackingNumber() != null &&
+            delivery.getIncoterm() != null &&
+            !(delivery.isHasDeparted() == false && delivery.isHasArrived() == true) &&
+            !delivery.getPartner().equals(ownPartnerEntity) &&
+            !ownPartnerEntity.getSites().stream().anyMatch(site -> site.getBpns().equals(delivery.getOriginBpns())) &&
```

Comment:
This is the opposite of what it should be. For OwnDelivery the origin has to be one of the sites of OwnPartnerEntity and destination needs to be a site of the Partner.